smac/intensification/parallel_successive_halving.py

```diff
@@ -67,15 +69,16 @@ class ParallelSuccessiveHalving(AbstractRacer):
         * highest_executed_budget - incumbent is the best in the highest budget run so far (default)
         * highest_budget - incumbent is selected only based on the highest budget
         * any_budget - incumbent is the best on any budget i.e., best performance regardless of budget
-    max_active_SH: int
-        Number of max active successive halvers to support
+    runner: DaskParallelRunner
```

Comment:
I'm still wondering if limiting the number of SH instances is really necessary. Is this to prevent the whole thing running havoc or does this provide functionality which I don't see right now?

Reply:
Yeah exactly, to prevent it to run havoc. I can think of certain troublesome scenarios... One example is, The tae takes 20 seconds to complete a job. So each time we query for sh instances, they all will be busy. So we will keep on creating new SH instances.

I believe the goal is to prevent having idle workers, then having more instances than number of workers doesn't make sense? Again this number is dynamically updated to the number of workers. But it is a mechanism to ensure we don't create new SH without a real purpose.

Reply:
Okay, got it. What do you think of passing the number of workers as part of `get_next_algorithm`? Then it's clearer that this function depends on the number of workers, the PSH doesn't need to know about the dask worker, and the facade doesn't need to know either.

Reply:
And other intensifiers which don't support parallelism can raise an exception if there is more than one worker.

Reply:
Thanks, this makes a lot of sense. I have coded and pushed this change